src/prioritizedTaskExecutor.ts

```diff
@@ -28,6 +28,7 @@ export class PrioritizedTaskExecutor {
    * @private
    * @param {Number} priority The priority of the task
    * @param {Function} fn The function that accepts the callback, which must be called upon the task completion.
+   * @returns {Promise}
```

Comment:
Why is this not async but returning a `Promise`?

Reply:
oops, thanks for the catch, will remove. i was playing with returning a promise but didn't need it at the end.